tests/unit_tests/arelle/test_validatexbrlcalcs.py

```diff
@@ -0,0 +1,140 @@
+
+    (1234.5678, None, -5, None, ('0', '1234.567800000000033833202906'), None),
+    (1234.5678, None, -1, None, ('1230', '4.567800000000033833202906'), None),
+    (1234.5678, None, 0, None, ('1234', '0.567800000000033833202906'), None),
+    (1234.5678, None, 1, None, ('1234.5', '0.067800000000033833202906'), None),
+    (1234.5678, None, 5, None, ('1234.5678', '0.000000000000033833202906'), None),
+
+    # precision + scale
+    (1234, -1, None, -1, None, None),
+    (1234, -1, None, 0, ('0', '1234'), None),
+    (1234, -1, None, 1, ('0', '12340'), None),
+    (1234, 0, None, -1, None, None),
+    (1234, 0, None, 0, None, None),
+    (1234, 0, None, 1, None, None),
+    (1234, 1, None, -5, None, None),
+    (1234, 1, None, -1, None, None),
+    (1234, 1, None, 0, ('1000', '234'), None),
+    (1234, 1, None, 1, ('12000', '340'), None),
+
+    (1234.5678, -1, None, -1, ('123.45', '0.0067800000000033833202906'), None),
+    (1234.5678, -1, None, 0, ('0', '1234.567800000000033833202906'), None),
+    (1234.5678, -1, None, 1, ('0', '12345.67800000000033833202906'), None),
+    (1234.5678, 0, None, -1, None, None),
+    (1234.5678, 0, None, 0, None, None),
+    (1234.5678, 0, None, 1, None, None),
+    (1234.5678, 1, None, -1, ('123.4567', '0.0000800000000033833202906'), None),
+    (1234.5678, 1, None, 0, ('1000', '234.567800000000033833202906'), None),
+    (1234.5678, 1, None, 1, ('12000', '345.67800000000033833202906'), None),
+
+    # decimals + scale
+    (1234, None, -1, -5, ('0', '0.01234'), None),
+    (1234, None, -1, -1, ('120', '3.4'), None),
+    (1234, None, -1, 0, ('1230', '4'), None),
+    (1234, None, -1, 1, None, None),
+    (1234, None, -1, 2, None, None),
+    (1234, None, 0, -5, ('0', '0.01234'), None),
+    (1234, None, 0, -1, ('123', '0.4'), None),
+    (1234, None, 0, 0, None, None),
+    (1234, None, 0, 1, None, None),
+    (1234, None, 1, -5, ('0', '0.01234'), None),
+    (1234, None, 1, -1, None, None),
+    (1234, None, 1, 0, None, None),
+    (1234, None, 1, 1, None, None),
+
+    (1234.5678, None, -1, -5, ('0', '0.01234567800000000033833202906'), None),
+    (1234.5678, None, -1, -1, ('120', '3.4567800000000033833202906'), None),
+    (1234.5678, None, -1, 0, ('1230', '4.567800000000033833202906'), None),
+    (1234.5678, None, -1, 1, ('12340', '5.67800000000033833202906'), None),
+    (1234.5678, None, 0, -5, ('0', '0.01234567800000000033833202906'), None),
+    (1234.5678, None, 0, -1, ('123', '0.4567800000000033833202906'), None),
+    (1234.5678, None, 0, 0, ('1234', '0.567800000000033833202906'), None),
+    (1234.5678, None, 0, 1, ('12345', '0.67800000000033833202906'), None),
+    (1234.5678, None, 1, -5, ('0', '0.01234567800000000033833202906'), None),
+    (1234.5678, None, 1, -1, ('123.4', '0.0567800000000033833202906'), None),
+    (1234.5678, None, 1, 0, ('1234.5', '0.067800000000033833202906'), None),
+    (1234.5678, None, 1, 1, ('12345.6', '0.07800000000033833202906'), None),
+
+    # large numbers
```

Comment:
"large precision" / "large decimals" might be clearer.  Additional tests of "large value" for whole/fractional parts might also be useful.